pdns/nod.cc

```diff
@@ -124,11 +138,12 @@ bool PersistentSBF::snapshotCurrent(std::thread::id tid)
     std::stringstream ss;
     ss << d_prefix << "_" << tid;
     f /= ss.str() + "_" + std::to_string(getpid()) + "." + bf_suffix;
+    path ftmp(unique_path(f.string() + ".%%%%%%%%"));
     if (exists(p) && is_directory(p)) {
       try {
         std::ofstream ofile;
         std::stringstream iss;
-        ofile.open(f.string(), std::ios::out | std::ios::binary);
+        ofile.open(ftmp.string(), std::ios::out | std::ios::binary);
```

Comment:
moving the `open` to after the `dump` shortens the window for leaving temp files around (even if we do clean them up)